src/lib/components/healthselectorv2/HealthSelector.component.js

```diff
@@ -0,0 +1,233 @@
+      border: 2px dashed ${defaultTheme.defaultTheme.darkRebrand.selectedActive};
+      background-color: ${defaultTheme.defaultTheme.darkRebrand
+        .buttonSecondary};
+    }
+    .sc-select__value-container {
+      overflow: visible;
+    }
+
+    &.sc-select__control--menu-is-open {
+      .sc-select__indicator {
+        transform: rotate(180deg);
+      }
+    }
+  }
+`;
+
+const icons = {
+  all: (
+    <ThreeCirclesWrapper>
+      <CSSCircle variant="statusHealthy" />
+      <CSSCircle variant="statusWarning" />
+      <CSSCircle variant="statusCritical" />
+    </ThreeCirclesWrapper>
+  ),
+  healthy: <Icon name={'Check-circle'} color={'statusHealthy'} size={'lg'} />,
+  warning: (
+    <Icon name={'Exclamation-circle'} color={'statusWarning'} size={'lg'} />
+  ),
+  critical: <Icon name={'Times-circle'} color={'statusCritical'} size={'lg'} />,
+  unknown: <Icon name={'Info'} color={'infoPrimary'} size={'lg'} />,
+};
+
+const optionsDefaultConfiguration: optionsConfiguration = {
+  all: {
+    icon: icons.all,
+    label: 'All healthy statuses',
+    shortLabel: 'All',
+    value: 'all',
+    isHidden: false,
+  },
+  healthy: {
+    icon: icons.healthy,
+    label: 'Healthy only',
+    shortLabel: 'Healthy',
+    value: 'healthy',
+    isHidden: false,
+  },
+  critical: {
+    icon: icons.critical,
+    label: 'Critical only',
+    shortLabel: 'Critical',
+    value: 'critical',
+    isHidden: false,
+  },
+  warning: {
+    icon: icons.warning,
+    label: 'Warning only',
+    shortLabel: 'Warning',
+    value: 'warning',
+    isHidden: false,
+  },
+  unknown: {
+    icon: icons.unknown,
+    label: 'Unknown only',
+    shortLabel: 'Unknown',
+    value: 'unknown',
+    disabled: false,
+  },
+};
+
+function HealthSelectorv2(props: Props) {
+  const {
+    optionsConfiguration,
+    label = 'Health',
+    onChange,
+    defaultValue = 'all',
+    ...rest
+  } = props;
+
+  const options = [];
+
+  Object.keys({
+    ...optionsDefaultConfiguration,
+    ...optionsConfiguration,
+  }).forEach((key) => {
+    /*
+    to replace with optionalChaining when available in core-ui
+    ...optionsDefaultConfiguration?.[key],
+    ...optionsConfiguration?.[key],
+    */
+    options.push({
+      ...optionsDefaultConfiguration[key],
+      ...(optionsConfiguration || {})[key],
+    });
+  });
+
+  const SingleValue = (props) => {
+    return (
+      <SingleValueWrapper>
+        <p data-testid="singleValueLabel">{label}</p>{' '}
+        <ValueIcon>{props.data.icon}</ValueIcon>
+        <p data-testid="singleValueShortLabel">
+          {props.data.optionProps.shortLabel}
+        </p>
+      </SingleValueWrapper>
+    );
+  };
+
+  return (
+    <SelectStyle
+      components={{ SingleValue }}
+      onChange={onChange}
+      defaultValue={defaultValue}
+      {...rest}
+    >
+      {options.map(
+        (optionValue, i) =>
+          !optionValue.isHidden && (
+            <Option
+              key={optionValue.value}
```

Comment:
If we display 2 health selector (which is one of our use cases !) on the same page react will complain about duplicated keys I think. We may need to prefix this key somehow, to be checked